operator/v1/types_ingress.go

```diff
@@ -698,7 +699,10 @@ type AWSClassicLoadBalancerParameters struct {
 }
 
 // AWSNetworkLoadBalancerParameters holds configuration parameters for an
-// AWS Network load balancer.
+// AWS Network load balancer. For Example: Setting AWS EIPs https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html
+// +kubebuilder:validation:XValidation:rule=`has(self.subnets) && has(self.subnets.ids) && has(self.subnets.names) && has(self.eipAllocations) ? size(self.subnets.ids + self.subnets.names) == size(self.eipAllocations) : true`,message="number of subnets must be equal to number of eipAllocations"
+// +kubebuilder:validation:XValidation:rule=`has(self.subnets) && has(self.subnets.ids) && !has(self.subnets.names) && has(self.eipAllocations) ? size(self.subnets.ids) == size(self.eipAllocations) : true`,message="number of subnets must be equal to number of eipAllocations"
+// +kubebuilder:validation:XValidation:rule=`has(self.subnets) && has(self.subnets.names) && !has(self.subnets.ids) && has(self.eipAllocations) ? size(self.subnets.names) == size(self.eipAllocations) : true`,message="number of subnets must be equal to number of eipAllocations"
```

Comment:
We missed this. Should be under `FeatureGateAwareXValidation`, but it's a bit problematic from the start because it should be under both `IngressControllerLBSubnetsAWS` AND `SetEIPForNLBIngressController`, because it requires both, and neither are Default on yet.

You will get errors when you create the CRD such as:
```
ERROR: <input>:1:30: undefined field 'subnets'
 | has(self.subnets) && has(self.subnets.ids) && has(self.subnets.names) && has(self.eipAllocations) ? size(self.subnets.ids + self.subnets.names) == size(self.eipAllocations) : true
 | .............................^
```

Fixing in https://github.com/openshift/api/pull/1966

Reply:
bumped https://github.com/openshift/api/pull/1966 in CIO PR https://github.com/openshift/cluster-ingress-operator/pull/1109 which is merged.